File: 5 - Coding Interview Workshops/Week1/Code/JavaScript/LinearSearch.js
// JavaScript Code
function linearSearch(arr, target) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === target) {
            return i;  // Return the index if the target is found
        }
    }
    return -1;  // Return -1 if the target is not found
}

// Example usage
const arr = [4, 2, 8, 5, 1, 9, 3];
const target = 1;
const result = linearSearch(arr, target);
if (result !== -1) {
    console.log(`Element found at index ${result}`);
} else {
    console.log("Element not found");
}

File: 5 - Coding Interview Workshops/Week01/Code/Python/LinearSearch.py
# Python Code
def linear_search(arr, target):
    for i in range(len(arr)):
        if arr[i] == target:
            return i  # Return the index if the target is found
    return -1  # Return -1 if the target is not found

# Example usage
arr = [4, 2, 8, 5, 1, 9, 3]
target = 1
result = linear_search(arr, target)
if result != -1:
    print("Element found at index {}.".format(result)) 
else:
    print("Element not found")